src/aggregate.js

```diff
@@ -34,29 +35,28 @@ const config = require('./config');
  *    -after {String} The _id to start querying the page.
  *    -before {String} The _id to start querying previous page.
  *    -options {Object} Aggregation options
+ * @param {(() => number) | undefined | null} getPipelineIndexFn Functor that determines where to insert the pagination-specific operations in the pipeline
+ *    This parameter controls the placement of the pagination-specific operations in the pipeline to allow more advanced use cases
+ *    - undefined => preserves legacy behavior (backwards-compatible)
+ *    - null => allows a more "common" behavior that allows data transformation prior to pagination (eg. includes reshaping _id value to be used in default pagination)
+ *    - custom function => allows for a user-defined impl (special/advanced cases such as post-processing of results server-side)
  */
-module.exports = async function aggregate(collection, params) {
+module.exports = async function aggregate(collection, params, getPipelineIndexFn) {
   params = _.defaults(await sanitizeParams(collection, params), { aggregation: [] });
   const cursorQuery = generateCursorQuery(params);
   const $sort = generateSort(params);
 
-  let index = _.findIndex(params.aggregation, (step) => !_.isEmpty(step.$match));
-
-  if (index < 0) {
-    params.aggregation.unshift({ $match: cursorQuery });
```

Comment:
My understanding of the existing behavior is that if a `$match` step isn't included `params.aggregation`, this code will add $match as the first step. Is this behavior preserved in the refactor?